core/src/main/scala/tech/beshu/ror/accesscontrol/factory/decoders/definitions/LdapServicesDecoder.scala

```diff
@@ -97,60 +98,60 @@ object LdapServicesDecoder {
     AsyncDecoderCreator
       .instance[LdapAuthorizationService] { c =>
-      val ldapServiceDecodingResult = for {
-        name <- c.downField("name").as[LdapService.Name]
-        connectionConfig <- connectionConfigDecoder(c)
-        userSearchFiler <- userSearchFilerConfigDecoder(c)
-        userGroupsSearchFilter <- userGroupsSearchFilterConfigDecoder(c)
-      } yield UnboundidLdapAuthorizationService.create(
-        name,
-        ldapConnectionPoolProvider,
-        connectionConfig,
-        userSearchFiler,
-        userGroupsSearchFilter
-      )
-      ldapServiceDecodingResult match {
-        case Left(error) => Task.now(Left(error))
-        case Right(task) => task.flatMap {
-          case Left(HostConnectionError(hosts)) =>
-            val connectionErrorMessage = Message(s"There was a problem with LDAP connection to: ${hosts.map(_.toString()).toList.mkString(",")}")
-            Task.now(Left(DecodingFailureOps.fromError(DefinitionsLevelCreationError(connectionErrorMessage))))
-          case Left(ServerDiscoveryConnectionError(recordName, providerUrl)) =>
-            val connectionErrorMessage = Message(s"There was a problem with LDAP connection in discovery mode. Connection details: recordName=${recordName.getOrElse("default")}, providerUrl=${providerUrl.getOrElse("default")}")
-            Task.now(Left(DecodingFailureOps.fromError(DefinitionsLevelCreationError(connectionErrorMessage))))
-          case Right(service) =>
-            Task.now(Right(service))
+        val ldapServiceDecodingResult = for {
+          name <- c.downField("name").as[LdapService.Name]
+          connectionConfig <- connectionConfigDecoder(c)
+          userSearchFiler <- userSearchFilerConfigDecoder(c)
+          userGroupsSearchFilter <- userGroupsSearchFilterConfigDecoder(c)
+        } yield UnboundidLdapAuthorizationService.create(
+          name,
+          ldapConnectionPoolProvider,
+          connectionConfig,
+          userSearchFiler,
+          userGroupsSearchFilter
+        )
+        ldapServiceDecodingResult match {
+          case Left(error) => Task.now(Left(error))
+          case Right(task) => task.flatMap {
+            case Left(HostConnectionError(hosts)) =>
+              val connectionErrorMessage = Message(s"There was a problem with LDAP connection to: ${hosts.map(_.toString()).toList.mkString(",")}")
+              Task.now(Left(DecodingFailureOps.fromError(DefinitionsLevelCreationError(connectionErrorMessage))))
+            case Left(ServerDiscoveryConnectionError(recordName, providerUrl)) =>
+              val connectionErrorMessage = Message(s"There was a problem with LDAP connection in discovery mode. Connection details: recordName=${recordName.getOrElse("default")}, providerUrl=${providerUrl.getOrElse("default")}")
+              Task.now(Left(DecodingFailureOps.fromError(DefinitionsLevelCreationError(connectionErrorMessage))))
```

Comment:
The private function would improve readability, e.g.`ConnectionError` => `DefinitionsLevelCreationError`